daemons/attrd/attrd_utils.c

```diff
@@ -34,44 +33,17 @@ static GMainLoop *mloop = NULL;
  */
 GHashTable *peer_protocol_vers = NULL;
 
-/*!
- * \internal
- * \brief  Set requesting_shutdown state
- */
-void
-attrd_set_requesting_shutdown(void)
-{
-    requesting_shutdown = true;
-}
-
-/*!
- * \internal
- * \brief  Clear requesting_shutdown state
- */
-void
-attrd_clear_requesting_shutdown(void)
-{
-    requesting_shutdown = false;
-}
-
 /*!
  * \internal
  * \brief Check whether local attribute manager is shutting down
  *
- * \param[in] if_requested  Also consider presence of "shutdown" attribute
- *
- * \return \c true if local attribute manager has begun shutdown sequence
- *         or (if \p if_requested is \c true) whether local node has a nonzero
- *         "shutdown" attribute set, otherwise \c false
- * \note Most callers should pass \c false for \p if_requested, because the
- *       attribute manager needs to continue performing while the controller is
- *       shutting down, and even needs to be eligible for election in case all
- *       nodes are shutting down.
+ * \return \c true if local attribute manager has begun shutdown sequence,
+ *         otherwise \c false
  */
 bool
-attrd_shutting_down(bool if_requested)
+attrd_shutting_down(void)
 {
-    return shutting_down || (if_requested && requesting_shutdown);
+    return shutting_down;
```

Comment:
At this point, we might as well kill the function and just use the flag everywhere instead.

Reply:
I like the flag being static. It ensures that all the proper shutdown stuff has been done if the flag is set to true.